test/support/analyzer_verification/suppress_if.ex

```diff
@@ -0,0 +1,59 @@
+defmodule ElixirAnalyzer.Support.AnalyzerVerification.SuppressIf do
+  @moduledoc """
+  This is an analyzer extension module to test the option :suppress_if between features/assert_call/common checks
+  """
+
+  alias ElixirAnalyzer.Constants
+  use ElixirAnalyzer.ExerciseTest
+
+  feature "feature 1: no foo() unless common check was found" do
+    find :none
+    comment "feature 1: foo() was called"
+    suppress_if Constants.solution_debug_functions(), :fail
+
+    form do
+      foo()
+    end
+  end
+
+  assert_no_call "assert 1: no foo() unless common check was found" do
+    comment "assert 1: foo() was called"
+    # This should work:
+    # suppress_if Constants.solution_debug_functions(), :fail
+    # but in assert_call, it gets treated as an AST, so the final string is required 
+    suppress_if "elixir.solution.debug_functions", :fail
```

Comment:
This is an annoying bug. I will open an issue for it.
I think it is related to that other bug that doesn't let you use
```elixir
alias alias ElixirAnalyzer.Constants
assert_no_call "x" do
  comment Constants.exercise()
...
```
but forces you to use the full path for the comment
```elixir
assert_no_call "x" do
  comment ElixirAnalyzer.Constants.exercise()
...
```

Reply:
I once tried to debug it but we're doing so much metaprogramming that I gave up. Anyway, it's good to open that issue.

Reply:
I had never really done any metaprogramming before contributing here... Talk about a baptism by fire!